corehq/apps/hqmedia/tasks.py

```diff
@@ -182,13 +181,24 @@ def build_application_zip(include_multimedia_files, include_index_files, app,
                     z.writestr(path, data, file_compression)
                     progress += file_progress / file_count
                     DownloadBase.set_progress(build_application_zip, progress, 100)
-                    if extension not in MULTIMEDIA_EXTENSIONS:
-                        file_cache[path] = data
 
-        errors.extend(find_missing_locale_ids_in_ccz(file_cache))
-
-        if include_index_files and include_multimedia_files:
-            errors.extend(check_ccz_multimedia_integrity(app.domain, fpath))
+        # Integrity check that all media files present in media_suite.xml were added to the zip
+        if include_multimedia_files and include_index_files and toggles.CAUTIOUS_MULTIMEDIA.enabled(app.domain):
+            with open(fpath, 'rb') as tmp:
+                with zipfile.ZipFile(tmp, "r") as z:
+                    media_suites = [f for f in z.namelist() if re.search(r'\bmedia_suite.xml\b', f)]
+                    if len(media_suites) != 1:
+                        message = _('Could not identify media_suite.xml in CCZ')
+                        errors.append(message)
+                    else:
+                        with z.open(media_suites[0]) as media_suite:
+                            from corehq.apps.app_manager.xform import parse_xml
+                            parsed = parse_xml(media_suite.read())
+                            resources = {node.text for node in
```

Comment:
Why it is to be a set not just a list. Also We are sure the order does not matter here?

Reply:
Because, presumably, a set would avoid catching duplicate errors in the lines that follow.